src/main/java/uk/gov/hmcts/reform/iahearingsapi/domain/mappers/CaseDataToServiceHearingValuesMapper.java

```diff
@@ -138,4 +146,55 @@ public String getRespondentName(AsylumCase asylumCase) {
             return "Entry Clearance Officer";
         }
     }
+
+    public String getListingCommentsFromHearingRequest(AsylumCase asylumCase) {
+        String responseFromHearingRequest = "";
+
+        String isVulnerabilitiesAllowed = asylumCase.read(IS_VULNERABILITIES_ALLOWED, String.class)
+            .orElse("");
+        String isMultimediaAllowed = asylumCase.read(IS_MULTIMEDIA_ALLOWED, String.class)
+            .orElse("");
+        String isAdditionalAdjustmentsAllowed = asylumCase.read(IS_ADDITIONAL_ADJUSTMENTS_ALLOWED, String.class)
+            .orElse("");
+
+        responseFromHearingRequest += getGrantedHearingResponseFromField(
+            asylumCase,
+            isVulnerabilitiesAllowed,
+            "Adjustments to accommodate vulnerabilities: ",
+            VULNERABILITIES_TRIBUNAL_RESPONSE
+        );
+
+        responseFromHearingRequest += getGrantedHearingResponseFromField(
+            asylumCase,
+            isMultimediaAllowed,
+            "Multimedia equipment: ",
+            MULTIMEDIA_TRIBUNAL_RESPONSE
+        );
+
+        responseFromHearingRequest += getGrantedHearingResponseFromField(
+            asylumCase,
+            isAdditionalAdjustmentsAllowed,
+            "Other adjustments: ",
+            ADDITIONAL_TRIBUNAL_RESPONSE
+        );
+
+        return responseFromHearingRequest;
+
+    }
+
+    private String getGrantedHearingResponseFromField(AsylumCase asylumCase,
+                                                      String isGrantedResponse,
+                                                      String hearingRequestTitle,
+                                                      AsylumCaseFieldDefinition responseField) {
+        String result = "";
+
+        if (GRANTED.getValue().equals(isGrantedResponse)) {
+            result += hearingRequestTitle;
```

Comment:
Would it be nicer to use String.format or StringBuilder here rather than concatenation?

Reply:
code updated and changed to StringBuilder, please help to review, thanks